### PR TITLE
Correcting scenario inconsistencies and output checks

### PR DESCRIPTION
**Pull request by tomoaki-ii**
This PR fixes discrepancies between the tutorial documentation and the actual output and changes the manifest to suit original scenario.

1. [The documented output at line 115-116](https://github.com/kubernetes/website/blob/ddec959814fff4d4d32bd401c9567c3b62a4d68f/content/en/docs/tutorials/stateful-application/basic-stateful-set.md?plain=1#L115) of `basic-stateful-set.md` shows DESIRED and CURRENT status columns for `kubectl get statefulset web`.

```
NAME      DESIRED   CURRENT   AGE
web       2         1         20s
```

 However, the actual output displays READY status instead. This is updated to match reality.

```
NAME   READY   AGE
web    2/2     37s
```

2. The documented container image names [at line 640-642](https://github.com/kubernetes/website/blob/ddec959814fff4d4d32bd401c9567c3b62a4d68f/content/en/docs/tutorials/stateful-application/basic-stateful-set.md?plain=1#L596) after patching statefulset by command `kubectl patch statefulset web --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/image", "value":"gcr.io/google_containers/nginx-slim:0.8"}]'` and getting pod images via  `for p in 0 1 2; do kubectl get pod "web-$p" --template '{{range $i, $c := .spec.containers}}{{$c.image}}{{end}}'; echo; done` are shown as `registry.k8s.io/nginx-slim:0.8`.
But the actual image names are `gcr.io/google_containers/nginx-slim:0.8`.                                                                        
<br>If you try to display this as `gcr.io/google_containers/nginx-slim:0.8`, it is not following  the community rules. Also, even if it is displayed as `registry.k8s.io/nginx-slim:0.8`, the original container image is `registry.k8s.io/nginx-slim:0.8`, so the container image will not be updated and will proceed. This contradicts the documentation.
<br>So I changed the container image in Statefulset in the first document (application/web/web.yaml) to `registry.k8s.io/nginx-slim:0.7` and rolled it to `registry.k8s.io/nginx-slim:0.8` I changed the format to update and updated it to a format that does not conflict with the scenario.

### CODE REVIEW

**Comment by sftim** [comment]
Thanks
/lgtm

File: content/en/docs/tutorials/stateful-application/basic-stateful-set.md [hunk 1]
```diff
@@ -637,9 +637,9 @@ Get the Pods to view their container images:
 for p in 0 1 2; do kubectl get pod "web-$p" --template '{{range $i, $c := .spec.containers}}{{$c.image}}{{end}}'; echo; done
 ```
 ```
-registry.k8s.io/nginx-slim:0.8
-registry.k8s.io/nginx-slim:0.8
-registry.k8s.io/nginx-slim:0.8
+gcr.io/google_containers/nginx-slim:0.8
+gcr.io/google_containers/nginx-slim:0.8
+gcr.io/google_containers/nginx-slim:0.8
```
Comment (atoato88) [inline content/en/docs/tutorials/stateful-application/basic-stateful-set.md:642]: @sftim @tomoaki-ii 
I think the `gcr.io` is replaced with `registry.k8s.io` on the community.
This updates does not follow the rules, I think.

WDYT?

Comment (atoato88) [reply content/en/docs/tutorials/stateful-application/basic-stateful-set.md:642]: At a same time, If we use `registry.k8s.io/nginx-slim:0.8` in the patch command [here](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#rolling-update), it does not create any changes in the manifest, right?

I'm confusing about the scenerio of this page. :thinking:

**Comment by sftim** [comment]
> https://github.com/kubernetes/website/pull/45878#discussion_r1568096446

Ah good point
/lgtm cancel
Can we use two different tags for the same image reference?

**Comment by tomoaki-ii** [comment]
Thanks
I have reviewed again what we originally wanted to correct, the comments I received and the flow of the tutorial.
I will propose improvements based on these points.

Change the container image of the Statefulset in the first document(application/web/web.yaml ) to registry.k8s.io/nginx-slim:0.7 and do RollingUpdate it to registry.k8s.io/nginx-slim:0.8 as per the document.

I would appreciate it if you could give me your opinion.

**Comment by atoato88** [comment]
Do you have any updates?
It may be good to update this PR to follow the suggestion.
Because it is easy to understand more.

**Comment by atoato88** [comment]
@tomoaki-ii 
Gentle ping.
I think your suggestion looks good.
Could you update this PR to follow your suggestion, and check the validity of scenario on this page?

**Comment by atoato88** [comment]
Deploy preview:
- https://deploy-preview-45878--kubernetes-io-main-staging.netlify.app/docs/tutorials/stateful-application/basic-stateful-set/

File: content/en/docs/tutorials/stateful-application/basic-stateful-set.md [hunk 2]
```diff
@@ -639,7 +639,7 @@ for p in 0 1 2; do kubectl get pod "web-$p" --template '{{range $i, $c := .spec.
 ```
 registry.k8s.io/nginx-slim:0.8
 registry.k8s.io/nginx-slim:0.8
-registry.k8s.io/nginx-slim:0.8
+registry.k8s.io/nginx-slim:0
```
Comment (atoato88) [inline content/en/docs/tutorials/stateful-application/basic-stateful-set.md:642]:
```suggestion
registry.k8s.io/nginx-slim:0.8
```
Is it typo?
I think the tag here is `0.8`.

**Comment by tomoaki-ii** [comment]
Thanks
I updated the container image and the contents of application/web/web.yaml according to what I presented last time.

It is working without any problems in my execution environment.
Please confirm.

I will later fix the text I originally wrote in line with this update.

**Comment by atoato88** [comment]
@sftim 
Could you check the update?

**Comment by shu-mutou** [comment]
@sftim @enisoc Sorry to bother you, but could you review this PR? Thanks!

**Comment by sftim** [comment]
(The suggested reviewers were set in https://github.com/kubernetes/website/pull/45878#event-12466526084)
I'll take a look

**Review by sftim — COMMENTED** [review]
Thanks

/lgtm
/approve

**Pull request merged**